### PR TITLE
Basic CRUD routes created + Basic migration.sql created

### PR DESCRIPTION
**Pull request by ethanberke**
CRUD routes are planned to be changed, just wanted to have basic routes available to build off of.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -16,12 +16,73 @@ const app = express();
...
 });
 
+app.post("/api/videogames", (req, res) => {
+  const { name, age_rating, genre, rating } = req.body;
+  client.query(
+    "INSERT INTO videogames(name, age_rating, genre, rating) VALUES($1, $2, $3, $4) RETURNING *",
+    [name, age_rating, genre, rating]
+    )
+    .then((data) => {
+      res.json(data.rows[0]);
```
Comment (gnprwx) [inline server/server.js:36]: Be sure to make this a 201 :)

File: server/migration.sql [hunk 2]
```diff
@@ -1,12 +1,11 @@
-DROP TABLE IF EXISTS tasks;
+DROP TABLE IF EXISTS videogames;
 
-CREATE TABLE tasks (
-  id SERIAL,
-  description TEXT
+CREATE TABLE videogames (
+  id SERIAL PRIMARY KEY,
+  name TEXT,
+  age_rating TEXT,
```
Comment (gnprwx) [inline server/migration.sql:6]: Should this be int?

File: server/server.js [hunk 3]
```diff
@@ -16,12 +16,73 @@ const app = express();
...
+    )
+      .then((data) => {
+        if (data.rows.length == 0){
+          res.sendStatus(404);
+        }
+        console.log("updated videogame: ", data.rows[0]);
+        res.send(data.rows[0]);
+      }).catch((err) => {
+        console.error(err);
+        res.sendStatus(500);
```
Comment (gnprwx) [inline server/server.js:65]: This could be a part of error middleware so each route can get it with next(err).

File: server/server.js [hunk 4]
```diff
@@ -16,12 +16,73 @@ const app = express();
...
+    [id])
+    .then((data) => {
+      if (data.rows.length === 0) {
+        res.sendStatus(404);
+      } else {
+        res.send(data.rows[0]);
+      }
+    }).catch((err) => {
+      console.error(err);
+      res.sendStatus(500);
```
Comment (gnprwx) [inline server/server.js:82]: Next middleware

File: server/migration.sql (hunk 2 above)
Comment (ethanberke) [reply server/migration.sql:6]: Are you talking about the age_rating?  I was thinking E, E10, T, M

File: server/server.js [hunk 5]
```diff
@@ -16,12 +16,73 @@ const app = express();
...
+  client.query(
+    "INSERT INTO videogames(name, age_rating, genre, rating) VALUES($1, $2, $3, $4) RETURNING *",
+    [name, age_rating, genre, rating]
+    )
+    .then((data) => {
+      res.json(data.rows[0]);
+    })
+    .catch((error) => {
+      console.error("Error creating videogame: ", error);
+      res.status(500).json({ error: "Internal server error" });
```
Comment (gnprwx) [inline server/server.js:40]: Next error middleware

Comment (ethanberke) [reply server/server.js:40]: I can adjust that

File: server/server.js [hunk 6]
```diff
@@ -16,12 +16,73 @@ const app = express();
...
-  client.query("SELECT * FROM tasks").then((rows) => {
+app.get("/", (req, res) => {
+  res.send("Steam Page");
+});
+
+app.get("/api/videogames", (req, res) => {
+  client.query("SELECT * FROM videogames").then((rows) => {
     res.send(rows);
   });
 });
```
Comment (gnprwx) [inline server/server.js:27]: Could add a next error middleware

**Review by gnprwx — COMMENTED** [review]
Some changes that can be implemented. Up to y'all if you wanna merge now or later.

File: server/migration.sql (hunk 2 above)
Comment (gnprwx) [reply server/migration.sql:6]: Ah gotcha. Sounds good!

File: server/server.js [hunk 7]
```diff
@@ -16,12 +16,73 @@ const app = express();
 
 app.use(express.json());
 
-app.get("/api/tasks", (req, res) => {
-  client.query("SELECT * FROM tasks").then((rows) => {
+app.get("/", (req, res) => {
```
Comment (gnprwx) [inline server/server.js:19]: This can probably be replaced with a catch-all 404 for anywhere that's not defined as an api route. Then we in business!

**Review by gnprwx — APPROVED** [review]

**Pull request merged**